src/Subsampling/include/gudhi/sparsify_point_set.h

```diff
@@ -14,13 +14,19 @@
 # include <boost/iterator/function_output_iterator.hpp>
 
 #include <gudhi/Kd_tree_search.h>
+#include <CGAL/version.h>  // for CGAL_VERSION_NR
 #ifdef GUDHI_SUBSAMPLING_PROFILING
 #include <gudhi/Clock.h>
 #endif
 
 #include <cstddef>
 #include <vector>
 
+// Make compilation fail - required for external projects
+#if CGAL_VERSION_NR < 1041101000
+# error subsampling::sparsify_point_set is only available with CGAL >= 4.11
```

Comment:
Even if this error will never be shown, I would prefer to keep it because removing it would imply to know exactly what happens for CGAL, and like that we keep consistency and knowledge between code and CMake conditions inside the gudhi code. Is it ok for you ?

Reply:
> Even if this error will never be shown, I would prefer to keep it because removing it would imply to know exactly what happens for CGAL,

DRY and locality both say this error makes more sense in the file that actually directly uses CGAL.

> and like that we keep consistency and knowledge between code and CMake conditions inside the gudhi code.

CMake stuff has much lower priority to me...
Do we still care about CGAL < 4.11 at all? Even Ubuntu 18.04 had 4.11.

> Is it ok for you ?

Yes, it is ok to keep it a bit longer, it has been here for a while anyway.

Reply:
> Yes, it is ok to keep it a bit longer, it has been here for a while anyway.

Uh, my brain must be off, this is new code added by this PR... I don't like adding this, but I won't fight against it.